src/auth/jsonfile.rs

```diff
@@ -0,0 +1,69 @@
+use crate::auth::*;
+
+use futures::Future;
+
+use std::fs;
+use serde::Deserialize;
+use serde_json;
+
+use tokio::timer::Delay;
+use std::time::{Duration, Instant};
+
+#[derive(Deserialize, Clone, Debug)]
+struct Credentials {
+    username: String,
+    password: String,
+}
+
+/// [`Authenticator`] implementation that authenticates against a JSON file.
+///
+/// [`Authenticator`]: ../trait.Authenticator.html
+#[derive(Clone, Debug)]
+pub struct JsonfileAuthenticator {
+    credentials_list: Vec<Credentials>,
+}
+
+impl JsonfileAuthenticator {
+    /// Initialize a new [`JsonfileAuthenticator`] from file.
+    pub fn new(filename: String) -> Self {
```

Comment:
More for interest sake: Instead of String you could use `Into<String>` to make it much more flexible for the caller to  pass in many types of strings. [Here](https://github.com/bolcom/libunftp/pull/124/commits/e52e4e36b6dfa848339fcbe9e66c24c5bac841cf) is an example

Reply:
I see what you mean. I didn't know this yet. I'll try to do this.